packages/integrations/src/homeassistant/homeassistant-integration.ts

```diff
@@ -53,21 +44,61 @@ export class HomeAssistantIntegration extends Integration {
    * @param entityId - The ID of the entity to toggle.
    * @returns A boolean indicating whether the toggle action was successful.
    */
-  async triggerToggleAsync(entityId: string) {
+  public async triggerToggleAsync(entityId: string) {
     try {
-      const response = await fetch(appendPath(this.integration.url, "/services/homeassistant/toggle"), {
-        headers: {
-          Authorization: `Bearer ${this.getSecretValue("apiKey")}`,
-        },
-        body: JSON.stringify({
-          entity_id: entityId,
-        }),
-        method: "POST",
+      const response = await this.postAsync("/api/services/homeassistant/toggle", {
+        entity_id: entityId,
       });
+
       return response.ok;
     } catch (err) {
       logger.error(`Failed to fetch from '${this.integration.url}': ${err as string}`);
       return false;
     }
   }
+
+  public async testConnectionAsync(): Promise<void> {
+    await super.handleTestConnectionResponseAsync({
+      queryFunctionAsync: async () => {
+        return await this.getAsync("/api/config");
+      },
+    });
+  }
+
+  /**
+   * Makes a GET request to the Home Assistant API.
+   * It includes the authorization header with the API key.
+   * @param path full path to the API endpoint
+   * @returns the response from the API
+   */
+  private async getAsync(path: `/api/${string}`) {
+    return await fetch(appendPath(this.integration.url, path), {
+      headers: this.getAuthHeaders(),
+    });
+  }
+
+  /**
+   * Makes a POST request to the Home Assistant API.
+   * It includes the authorization header with the API key.
+   * @param path full path to the API endpoint
+   * @param body the body of the request
+   * @returns the response from the API
+   */
+  private async postAsync(path: `/api/${string}`, body: Record<string, string>) {
+    return await fetch(appendPath(this.integration.url, path), {
+      headers: this.getAuthHeaders(),
+      body: JSON.stringify(body),
+      method: "POST",
+    });
+  }
```

Comment:
**NIT**
For me these functions do not bring much value. They abstract business logic into a function which makes it harder to read. Additionally abstraction seems not required here to me because the code is already very de-duplicated. You can leave this as it is but I would be careful with too much abstraction (you have to read the JSDoc to see what happens in this function)

Reply:
IMO it makes it easier to use it and it's more centralized which allows to change less code when something changes